c/tests/test_stats.c

```diff
@@ -477,6 +477,30 @@ verify_node_general_stat_errors(tsk_treeseq_t *ts)
 }
 
 
+typedef int one_way_weighted_method(tsk_treeseq_t *self,
+        tsk_size_t num_weights, double *weights,
+        tsk_size_t num_windows, double *windows, double *result, tsk_flags_t options);
+
+static void
+verify_one_way_weighted_func_errors(tsk_treeseq_t *ts, one_way_weighted_method *method)
+{
+    // we don't have any specific errors for this function
+    // but we might add some in the future
+    int ret;
+    tsk_size_t num_samples = tsk_treeseq_get_num_samples(ts);
+    double *weights = malloc(num_samples * sizeof(double));
+    double result;
+    size_t j;
+
+    for (j = 0; j < num_samples; j++) {
+        weights[j] = 1.0;
+    }
+
+    ret = method(ts, 0, weights, 0, NULL, &result, 0);
+    CU_ASSERT_EQUAL_FATAL(ret, TSK_ERR_BAD_STATE_DIMS);
```

Comment:
Leaking memory here - need to free weights.

Reply:
heh. oh, right.